src/amm.rs

```diff
@@ -34,25 +35,36 @@ macro_rules! to_u128 {
     };
```

Comment:
Change macro to_u128 to be Balance::try_from. Perhaps we can also change to_u128 to to_balance, so it is clear.